x-pack/plugin/vectors/src/main/resources/org/elasticsearch/xpack/vectors/query/org.elasticsearch.xpack.vectors.txt

```diff
@@ -11,6 +11,40 @@ class org.elasticsearch.xpack.vectors.query.DenseVectorScriptDocValues {
 class org.elasticsearch.script.ScoreScript @no_import {
 }
 
+class org.elasticsearch.xpack.vectors.query.DenseVector {
+    DenseVector EMPTY
+    float getMagnitude()
+    # handle List<Number> and float[] arguments
+    double dotProduct(Object)
+    double l1Norm(Object)
+    double l2Norm(Object)
+    float[] getVector()
+    boolean isEmpty()
+    int getDims()
+    int size()
+}
+
+# implementation of DenseVector
+class org.elasticsearch.xpack.vectors.query.BinaryDenseVector {
+}
+
+# implementation of DenseVector
+class org.elasticsearch.xpack.vectors.query.KnnDenseVector {
+}
+
+class org.elasticsearch.xpack.vectors.query.DenseVectorDocValuesField {
+    DenseVector get()
+    DenseVector get(DenseVector)
+}
+
+# implementation of DenseVectorDocValuesField
+class org.elasticsearch.xpack.vectors.query.KnnDenseVectorDocValuesField {
+}
+
+# implementation of DenseVectorDocValuesField
+class org.elasticsearch.xpack.vectors.query.BinaryDenseVectorDocValuesField {
+}
+
```

Comment:
While not necessary as these are top-level field classes, for consistency we have duck-typed all of them so far.